nsxt/policy_utils.go

```diff
@@ -421,21 +448,47 @@ func nsxtPolicyPathResourceImporter(d *schema.ResourceData, m interface{}) ([]*s
 	return rd, nil
 }
 
-// This importer function accepts policy path only as import ID
-func nsxtPolicyPathOnlyResourceImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	return nsxtPolicyPathResourceImporterHelper(d, m)
+func getMultitenancyPathExample(pathExample string) string {
+	return fmt.Sprintf("%s or /orgs/[org]/projects/[project]%s", pathExample, pathExample)
 }
 
-func nsxtVPCPathResourceImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
-	if err != nil {
-		return rd, err
+// This importer function accepts policy path or resource ID
+func getFriendlyPolicyPathOrIDResourceImporter(pathExample string) func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	return func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+		rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
+		if errors.Is(err, ErrUnexpectedPolicyPath) {
+			return rd, fmt.Errorf("Invalid policy path %s; expected format: %s", d.Id(), pathExample)
+		}
+		if errors.Is(err, ErrNotAPolicyPath) {
+			if !isValidResourceID(d.Id()) {
+				return rd, fmt.Errorf("Invalid policy path or id %s; expected path format: %s", d.Id(), pathExample)
+			}
+			return rd, nil
+		} else if err != nil {
+			return rd, err
+		}
+		return rd, nil
 	}
-	projectID, vpcID := getContextDataFromSchema(d)
-	if projectID == "" || vpcID == "" {
-		return rd, fmt.Errorf("imported resource policy path should have both project_id and vpc_id fields")
+}
+
+// This importer function accepts VPC path only
+func getFriendlyVpcPathResourceImporter(pathExample string) func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	return func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+		rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
+		if errors.Is(err, ErrNotAPolicyPath) || errors.Is(err, ErrUnexpectedPolicyPath) {
+			return rd, fmt.Errorf("Invalid policy path %s; expected format: %s", d.Id(), pathExample)
+		}
+
+		if err != nil {
+			return rd, err
+		}
+
+		projectID, vpcID := getContextDataFromSchema(d)
```

Comment:
For code readability, can you add a comment stating that we check the context here because it's supposed to be filled by nsxtPolicyPathResourceImporterHelper